src/hotspot/share/gc/shared/threadLocalAllocBuffer.inline.hpp

```diff
@@ -41,11 +41,7 @@ inline HeapWord* ThreadLocalAllocBuffer::allocate(size_t size) {
   if (pointer_delta(end(), obj) >= size) {
     // successful thread-local allocation
 #ifdef ASSERT
-    // Skip mangling the space corresponding to the object header to
-    // ensure that the returned space is not considered parsable by
-    // any concurrent GC thread.
-    size_t hdr_size = oopDesc::header_size();
-    Copy::fill_to_words(obj + hdr_size, size - hdr_size, badHeapWordVal);
+    Copy::fill_to_words(obj, size, badHeapWordVal);
```

Comment:
So, this is safe after CMS removal, or for some other reason? I think this and other two removals near `Skip mangling the space corresponding to the object header` comment should be done in a separate PR to avoid cluttering this one.

Reply:
https://github.com/openjdk/jdk/commit/0917ad432eb3d01c104f03973c5c7ff52c6dfefe suggests skipping the header during the zapping was added for G1. So why it is safe to remove?